server/src/main/java/org/elasticsearch/index/query/QueryRewriteContext.java

```diff
@@ -258,4 +265,21 @@ public void onFailure(Exception e) {
     public Index getFullyQualifiedIndex() {
         return fullyQualifiedIndex;
     }
+
+    /**
+     * Returns the index settings for this context. This might return null if the
+     * context has not index scope.
+     */
+    public IndexSettings getIndexSettings() {
```

Comment:
I think I already extracted these changes in my refactoring...maybe the PR just needs update.